src/ucp/proto/proto_common.c

```diff
@@ -661,16 +659,21 @@ ucp_proto_common_init_caps(const ucp_proto_common_init_params_t *params,
         range1             = &caps->ranges[1];
         range1->max_length = params->max_length;
 
-        perf_multi = ucp_proto_common_ppln3_perf(send_ovrh, xfer, recv_ovrh,
-                                                 frag_size);
-
         /* Overhead of sending one fragment before starting the pipeline */
-        frag_ovrh = ucs_linear_func_apply(range0->perf, frag_size) -
-                    ucs_linear_func_apply(perf_multi, frag_size);
+        frag_ovrh =
+                ucs_linear_func_apply(range0->perf[UCP_PROTO_PERF_TYPE_SINGLE],
+                                      frag_size) -
+                ucs_linear_func_apply(range0->perf[UCP_PROTO_PERF_TYPE_MULTI],
+                                      frag_size);
 
         /* Apply the pipelining effect when sending multiple fragments */
-        range1->perf = ucs_linear_func_add(perf_multi,
-                                           ucs_linear_func_make(frag_ovrh, 0));
+        range1->perf[UCP_PROTO_PERF_TYPE_SINGLE] =
+                ucs_linear_func_add(range0->perf[UCP_PROTO_PERF_TYPE_MULTI],
+                                    ucs_linear_func_make(frag_ovrh, 0));
+
```

Comment:
should not be sending multiple fragments be the same as sending multiple messages (i.e. pipelined)?

Reply:
for case of "single", since we count latency, it also adds the overhead of sending the first fragment. for "multi" we never count this overhead